pkg/storage/tsdb/block/block.go

```diff
@@ -42,6 +43,14 @@ const (
 	DebugMetas = "debug/metas"
 )
 
+type ErrMarkerExists struct {
+	message string
+}
+
+func (e *ErrMarkerExists) Error() string {
```

Comment:
```suggestion
func (e ErrMarkerExists) Error() string {
```
Should not be a pointer, as it makes it's usage cumbersome.
Note that if it's a pointer, then the above call `errors.As(err, &block.ErrMarkerExists{})` will never be true.